lib/src/dom.dart

```diff
@@ -0,0 +1,356 @@
+      params['outlineColor'] = outlineColor;
+    }
+
+    return _sendCommand(new _Event('DOM.highlightRect', params));
+  }
+
+  Future<int> moveTo(int nodeId, int targetNodeId,
+      {int insertBeforeNodeId}) async {
+    var params = {'nodeId': nodeId, 'targetNodeId': targetNodeId};
+
+    if (insertBeforeNodeId != null) {
+      params['insertBeforeNodeId'] = insertBeforeNodeId;
+    }
+
+    var resp = await _sendCommand(new _Event('DOM.moveTo', params));
+    return resp.result['nodeId'];
+  }
+
+  Future<int> querySelector(int nodeId, String selector) async {
+    var resp = await _sendCommand(new _Event(
+        'DOM.querySelector', {'nodeId': nodeId, 'selector': selector}));
+    return resp.result['nodeId'];
+  }
+
+  Future<List<int>> querySelectorAll(int nodeId, String selector) async {
+    var resp = await _sendCommand(new _Event(
+        'DOM.querySelectorAll', {'nodeId': nodeId, 'selector': selector}));
+    return resp.result['nodeIds'];
+  }
+
+  Future removeAttribute(int nodeId, String name) => _sendCommand(
+      new _Event('DOM.removeAttribute', {'nodeId': nodeId, 'name': name}));
+  Future removeNode(int nodeId) =>
+      _sendCommand(new _Event('DOM.removeNode', {'nodeId': nodeId}));
+  Future requestChildNodes(int nodeId) =>
+      _sendCommand(new _Event('DOM.requestChildNodes', {'nodeId': nodeId}));
+  Future<int> requestNode(String objectId) async {
+    var resp = await _sendCommand(
+        new _Event('DOM.requestNode', {'objectId': objectId}));
+    return resp.result['nodeId'];
+  }
+
+  Future<WipRemoteObject> resolveNode(int nodeId, {String objectGroup}) async {
+    var params = {'nodeId': nodeId};
+    if (objectGroup != null) {
+      params['objectGroup'] = objectGroup;
+    }
+
+    var resp = await _sendCommand(new _Event('DOM.resolveNode', params));
+    return new WipRemoteObject(resp.result['object']);
+  }
+
+  Future setAttributeValue(int nodeId, String name, String value) =>
+      _sendCommand(new _Event('DOM.setAttributeValue', {
+    'nodeId': nodeId,
+    'name': name,
+    'value': value
+  }));
+  Future setAttributesAsText(int nodeId, String text, {String name}) {
+    var params = {'nodeId': nodeId, 'text': text};
+    if (name != null) {
+      params['name'] = name;
+    }
+    _sendCommand(new _Event('DOM.setAttributeValue', params));
+  }
+
+  Future<int> setNodeName(int nodeId, String name) async {
+    var resp = await _sendCommand(
+        new _Event('DOM.setNodeName', {'nodeId': nodeId, 'name': name}));
+    return resp.result['nodeId'];
+  }
+
+  Future setNodeValue(int nodeId, String value) => _sendCommand(
+      new _Event('DOM.setNodeValue', {'nodeId': nodeId, 'value': value}));
+
+  Future setOuterHtml(int nodeId, String outerHtml) => _sendCommand(new _Event(
+      'DOM.setOuterHTML', {'nodeId': nodeId, 'outerHtml': outerHtml}));
+
+  StreamController<AttributeModifiedEvent> _attributeModifiedController =
+      new StreamController<AttributeModifiedEvent>.broadcast();
```

Comment:
I like the RHS type info here as it could find errors when run in checked-mode.

I actually don't think the LHS-type info on these private fields add much, so am removing it.